spec/models/agents/tumblr_publish_agent_spec.rb

```diff
@@ -1,38 +1,87 @@
 require 'spec_helper'
 
 describe Agents::TumblrPublishAgent do
-  before do
-    @opts = {
-      :blog_name => "huginnbot.tumblr.com",
-      :post_type => "text",
-      :expected_update_period_in_days => "2",
-      :options => {
-        :title => "{{title}}",
-        :body => "{{body}}",
-      },
-    }
-
-    @checker = Agents::TumblrPublishAgent.new(:name => "HuginnBot", :options => @opts)
-    @checker.service = services(:generic)
-    @checker.user = users(:bob)
-    @checker.save!
-
-    @event = Event.new
-    @event.agent = agents(:bob_weather_agent)
-    @event.payload = { :title => "Gonna rain...", :body => 'San Francisco is gonna get wet' }
-    @event.save!
-
-    stub.any_instance_of(Agents::TumblrPublishAgent).tumblr {
-      stub!.text(anything, anything) { { "id" => "5" } }
-    }
+  describe "Should create post" do
+    before do
+      @opts = {
+        :blog_name => "huginnbot.tumblr.com",
+        :post_type => "text",
+        :expected_update_period_in_days => "2",
+        :options => {
+          :title => "{{title}}",
+          :body => "{{body}}",
+        },
+      }
+
+      @checker = Agents::TumblrPublishAgent.new(:name => "HuginnBot", :options => @opts)
+      @checker.service = services(:generic)
+      @checker.user = users(:bob)
+      @checker.save!
+
+      @event = Event.new
+      @event.agent = agents(:bob_weather_agent)
+      @event.payload = { :title => "Gonna rain...", :body => 'San Francisco is gonna get wet' }
+      @event.save!
+
+      @post_body = {
+        "id" => 5,
+        "title" => "Gonna rain...",
+        "link" => "http://huginnbot.tumblr.com/gonna-rain..."
+      }
+      stub.any_instance_of(Agents::TumblrPublishAgent).tumblr {
+        obj = Object.new
```

Comment:
if someone with more experience with `rr` and `rspec` could help me how to multiple stubbing that would be amazing.

```
stub!.text
stub!.posts
``` 
was throwing `NoMethodError text`. :(((